mdatagen: metrics.go.tmpl broken filter import

**Describe the bug**
Running the current mdatagen against receivers in collector-contrib directly breaks tests due to removal of the collector/filter import in

https://github.com/open-telemetry/opentelemetry-collector/pull/12092/files#diff-27825b678880d103cd17473d56b87f9fe90fe62f67eae7180bb85186e8742e32L19

**Steps to reproduce**
Here's an example for elasticsearch but seems to happen on other receivers as well.

```
$ cd opentelemetry-collector-contrib/receiver/elasticsearchreceiver/
$ git diff
$ mdatagen -v
mdatagen version v0.119.0
$ git diff internal/metadata/generated_metrics.go
diff --git a/receiver/elasticsearchreceiver/internal/metadata/generated_metrics.go b/receiver/elasticsearchreceiver/internal/metadata/generated_metrics.go
index 62fb59e3a8..0ca4bf71fb 100644
--- a/receiver/elasticsearchreceiver/internal/metadata/generated_metrics.go
+++ b/receiver/elasticsearchreceiver/internal/metadata/generated_metrics.go
@@ -6,7 +6,6 @@ import (
        "time"
 
        "go.opentelemetry.io/collector/component"
-       "go.opentelemetry.io/collector/filter"
        "go.opentelemetry.io/collector/pdata/pcommon"
        "go.opentelemetry.io/collector/pdata/pmetric"
        "go.opentelemetry.io/collector/receiver"
$ make test
...
# github.com/open-telemetry/opentelemetry-collector-contrib/receiver/elasticsearchreceiver/internal/metadata
internal/metadata/generated_metrics.go:5305:77: undefined: filter
internal/metadata/generated_metrics.go:5306:77: undefined: filter
...
```

**What did you expect to see?**
tests continue to pass

**What did you see instead?**
tests fail

**What version did you use?**
otel-collector-contrib@HEAD
mdatagen 0.119.0, reverting to 0.118.0 has the same issue, 0.117.0 worked as expected

**What config did you use?**
metadata file in elasticsearch receiver otel collector contrib

**Environment**
Ubuntu 20.04

**Additional context**
Contrib collector is still building, and those filter imports are still there, so wholly possible my setup is broken somehow - but at a basic level I don't see how removing the filter import from the template and relying on it later in the file is correct.

--- a/.chloggen/fix-12298.yaml
+++ b/.chloggen/fix-12298.yaml
@@ -1,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: mdatagen
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix broken imports in the generated files.
+
+# One or more tracking issues or pull requests related to the change
+issues: [12298]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [api]

--- a/cmd/mdatagen/internal/templates/component_test.go.tmpl
+++ b/cmd/mdatagen/internal/templates/component_test.go.tmpl
@@ -35,7 +35,6 @@ import (
 	"go.opentelemetry.io/collector/receiver/receivertest"
 	{{- end }}
 	{{- if isScraper }}
-	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.opentelemetry.io/collector/scraper"
 	"go.opentelemetry.io/collector/scraper/scrapertest"
 	{{- end }}

--- a/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
+++ b/cmd/mdatagen/internal/templates/telemetrytest.go.tmpl
@@ -3,7 +3,6 @@
 package {{ .Package }}test
 
 import (
-	"context"
 	"testing"
 
 	"github.com/stretchr/testify/require"

--- a/cmd/mdatagen/internal/templates/telemetrytest_test.go.tmpl
+++ b/cmd/mdatagen/internal/templates/telemetrytest_test.go.tmpl
@@ -3,14 +3,15 @@
 package {{ .Package }}test
 
 import (
-    "context"
+	"context"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/metric"
 	"go.opentelemetry.io/otel/sdk/metric/metricdata"
 	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
 
-    "go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/component/componenttest"
 )
 
 func TestSetupTelemetry(t *testing.T) {

